src/osd/PrimaryLogPG.cc

```diff
@@ -4241,12 +4241,17 @@ int PrimaryLogPG::do_tmapup(OpContext *ctx, bufferlist::iterator& bp, OSDOp& osd
   return result;
 }
 
-static int check_offset_and_length(uint64_t offset, uint64_t length, uint64_t max)
+static int check_offset_and_length(uint64_t offset, uint64_t length,
+  uint64_t max, DoutPrefixProvider *dpp)
 {
   if (offset >= max ||
       length > max ||
-      offset + length > max)
+      offset + length > max) {
+    ldpp_dout(dpp, 10) << __func__ << " "
+      << "osd_max_object_size >= 4GB; Hard limit of object size is 4GB.\n"
```

Comment:
no need to print the trailing `\n`.
nit, you are hardwiring the max to 4GB, but it is configurable.